src/main/java/org/jenkinsci/plugins/pollscm/SCMPollPublisher.java

```diff
@@ -0,0 +1,102 @@
+package org.jenkinsci.plugins.pollscm;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.AbstractProject;
+import hudson.model.Descriptor.FormException;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.BuildStepMonitor;
+import hudson.tasks.Notifier;
+import hudson.tasks.Publisher;
+import hudson.triggers.SCMTrigger;
+import hudson.triggers.Trigger;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import jenkins.tasks.SimpleBuildStep;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+public class SCMPollPublisher
+  extends Notifier
+  implements SimpleBuildStep
+{
+  @DataBoundConstructor
+  public SCMPollPublisher()
+  {
+    super();
+  }
+
+  public boolean needsToRunAfterFinalized()
+  {
+    return true;
+  }
+  
+  public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull Launcher launcher, @Nonnull TaskListener taskListener)
+    throws InterruptedException, IOException
+  {
+    AbstractProject project = (AbstractProject)run.getParent();
+    
+    Map<Thread, StackTraceElement[]> map = Thread.getAllStackTraces();
+    for (Thread thread : map.keySet()) {
+      if (thread.getName().contains("SCM polling"))
+      {
+        taskListener.getLogger().println("Contains scm polling.");
+        if (thread.getName().contains(project.getName()))
+        {
+          taskListener.getLogger().println("Interrupting thread.");
+          thread.interrupt();
```

Comment:
This is not a sufficient guarantee you are killing the right thread. Dangerous.

Also, when the trigger is already running, what is the point to kill it and start it again?